components/giscus-comments.tsx

```diff
@@ -0,0 +1,23 @@
+import Giscus from "@giscus/react";
+
+export default function Comment() {
+  return (
+  <>
+    <br />
+    <hr />
+    <br />
+    {process.env.NEXT_PUBLIC_REPO && process.env.NEXT_PUBLIC_REPO_ID && process.env.NEXT_PUBLIC_CATEGORY_ID ? (
+    <Giscus
+      repo= {process.env.NEXT_PUBLIC_REPO as `${string}/${string}`}
+      repoId= {process.env.NEXT_PUBLIC_REPO_ID}
+      category="Announcements"
+      categoryId= {process.env.NEXT_PUBLIC_CATEGORY_ID}
+      mapping="pathname"
+      reactionsEnabled="1"
+      emitMetadata="0"
+      theme="dark"
```

Comment:
Can we set this dynamically? Example:
https://github.com/WATonomous/watcloud-website/blob/f9c283fb1a8bec0d1da82a882a2d36a2927e398c/components/ui/sonner.tsx#L11

Reply:
Do you have any context as to how the `darkMode` variable is set? I am trying to do this but it seems to always be set as `true`. Also, would something like a `useEffect()` be able to update the comments component to change theme even after the page is rendered?

Reply:
This variable should be set according to the setting near the footer near every page:
<img width="1400" alt="image" src="https://github.com/WATonomous/watcloud-website/assets/5977478/ad1ab742-6343-4f36-a19f-ab7a566348cb">

Reply:
Yeah, even if I load the page with light mode the variable is still showing up as `true` not sure why

Reply:
Interesting, perhaps try this approach instead? I do remember one of these approaches didn't work as expected during development, but didn't look too deeply into it.

https://github.com/WATonomous/watcloud-website/blob/f9c283fb1a8bec0d1da82a882a2d36a2927e398c/components/assets.tsx#L202-L203

Reply:
This seemed to work. I'll create a new PR once I have access to the infra-config repo. Thanks for the help!